WordPress/src/main/java/org/wordpress/android/widgets/WPDialogSnackbar.java

```diff
@@ -0,0 +1,157 @@
+package org.wordpress.android.widgets;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+import android.support.design.widget.Snackbar;
+import android.support.design.widget.Snackbar.SnackbarLayout;
+import android.text.TextUtils;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.TextView;
+
+import org.wordpress.android.R;
+
+/**
+ * {@link Snackbar} with {@link android.app.Dialog}-like layout.  The layout views include title, message,
+ * positive button, and negative button.  Any empty or null view is hidden.  The only required view is message.
+ */
+public class WPDialogSnackbar {
+    private Snackbar mSnackbar;
+    private View mContentView;
+
+    private WPDialogSnackbar(@NonNull View view, @NonNull CharSequence text, int duration) {
+        switch (duration) {
+            case Snackbar.LENGTH_INDEFINITE:
+                mSnackbar = Snackbar.make(view, "", Snackbar.LENGTH_INDEFINITE);
+                break;
+            case Snackbar.LENGTH_LONG:
+                mSnackbar = Snackbar.make(view, "", Snackbar.LENGTH_LONG);
+                break;
+            default:
+                mSnackbar = Snackbar.make(view, "", Snackbar.LENGTH_SHORT);
+        }
+
+        // Set underlying snackbar layout.
+        SnackbarLayout snackbarLayout = (SnackbarLayout) mSnackbar.getView();
+        snackbarLayout.setPadding(0, 0, 0, 0);
+
+        // Hide underlying snackbar text and action.
+        TextView snackbarText = snackbarLayout.findViewById(android.support.design.R.id.snackbar_text);
+        snackbarText.setVisibility(View.INVISIBLE);
+        TextView snackbarAction = snackbarLayout.findViewById(android.support.design.R.id.snackbar_action);
+        snackbarAction.setVisibility(View.INVISIBLE);
+
+        mContentView = ((LayoutInflater) view.getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE))
+                .inflate(R.layout.dialog_snackbar, null);
+
+        TextView message = mContentView.findViewById(R.id.message);
+
+        // Hide message view when text is empty.
+        if (TextUtils.isEmpty(text)) {
+            message.setVisibility(View.GONE);
+        } else {
+            message.setVisibility(View.VISIBLE);
+            message.setText(text);
+        }
+
+        snackbarLayout.addView(mContentView, 0);
+    }
+
+    public void dismiss() {
+        if (mSnackbar != null) {
+            mSnackbar.dismiss();
+        }
+    }
+
+    public boolean isShowing() {
+        return mSnackbar != null && mSnackbar.isShown();
+    }
+
+    public static WPDialogSnackbar make(@NonNull View view, @NonNull CharSequence text, int duration) {
+        return new WPDialogSnackbar(view, text, duration);
+    }
+
+    public WPDialogSnackbar setNegativeButton(CharSequence text, final View.OnClickListener listener) {
```

Comment:
I am curious about the decision to return the `WPDialogSnackbar` from these 4 methods. Since we are changing the current instance, is there a benefit to it that I am missing?

Reply:
I wanted to keep it as similar to `Snackbar` as possible by allowing us to concatenate the `setNegativeButton`, `setNeutralButton`, `setPositiveButton`, and `setTitle` assignments when creating a new `WPDialogSnackbar`.  For example, we can do this:
```
 WPDialogSnackbar.make(view, "Message", Snackbar.LENGTH_INDEFINITE)
                .setTitle("Title")
                .setNeutralButton("Neutral", new OnClickListener() {
                    @Override public void onClick(View view) {
                        Toast.makeText(getActivity(), "Neutral tapped", Toast.LENGTH_SHORT).show();
                    }
                })
                .setNegativeButton("Negative", new OnClickListener() {
                    @Override public void onClick(View view) {
                        Toast.makeText(getActivity(), "Negative tapped", Toast.LENGTH_SHORT).show();
                    }
                })
                .setPositiveButton("Positive", new OnClickListener() {
                    @Override public void onClick(View view) {
                        Toast.makeText(getActivity(), "Positive tapped", Toast.LENGTH_SHORT).show();
                    }
                })
                .show();
            }
        });
```
If we don't return `WPDialogSnackbar`, we would have to do something like this to accomplish the same as above.
```
WPDialogSnackbar wpDialogSnackbar = WPDialogSnackbar.make(view, "Message", Snackbar.LENGTH_INDEFINITE);
wpDialogSnackbar.setTitle("Title");
wpDialogSnackbar.setNeutralButton("Neutral", new OnClickListener() {
    @Override public void onClick(View view) {
        Toast.makeText(getActivity(), "Neutral tapped", Toast.LENGTH_SHORT).show();
    }
});
wpDialogSnackbar.setNegativeButton("Negative", new OnClickListener() {
    @Override public void onClick(View view) {
        Toast.makeText(getActivity(), "Negative tapped", Toast.LENGTH_SHORT).show();
    }
});
wpDialogSnackbar.setPositiveButton("Positive", new OnClickListener() {
    @Override public void onClick(View view) {
        Toast.makeText(getActivity(), "Positive tapped", Toast.LENGTH_SHORT).show();
    }
});
wpDialogSnackbar.show();
```

Reply:
Makes sense - thanks for clarifying!